common/workunit/workunit.cpp

```diff
@@ -2620,6 +2620,40 @@ cost_type aggregateCost(const IConstWorkUnit * wu, const char *scope, bool exclu
     }
 }
 
+// aggregate disk costs from top-level subgraphs (when scope specified) or workflows (scope not specified)
+cost_type aggregateDiskAccessCost(const IConstWorkUnit * wu, const char *scope)
+{
+    WuScopeFilter filter;
+    if (!isEmptyString(scope))
+        filter.addScope(scope);
+    else
+        filter.addScope("");      // Needed to match scope
+    // subgraphs at nesting level 2. (Costs from child graphs and activities should have been summed up to subgraph level already)
+    // workflow at nesting level 1 but use 2 for workflow too so that if there is no workflow disk cost (yet), use subgraph disk costs instead
```

Comment:
This comment isn't quite right (I may have got this wrong in our discussion).  A global match, with nesting(2) would also include workflow and graphs, but not subgraphs.  A match on a workflow with nesting of 2 would include graphs and subgraphs.
You could use (3 - queryScopeDepth(scope)), although if so the function should be fixed to also treat null the same as ""

Reply:
I've updated the comment.  Hopefully, the new comment is more accurate.